File: backend/level1/Rental.py
from datetime import datetime
class Rental_price:
    def __init__(self,car_data,rental_data):
        #retrieving the dictionary in cars which has the same id as the car_id from rentals
        dictionary = list(filter(lambda car_data: car_data['id'] == rental_data["car_id"], car_data))[0]
        print(dictionary)
        self.id = rental_data["id"]
        #finding the next set of values needed in the price formula
        self.price_per_km = dictionary["price_per_km"]
        self.price_per_day = dictionary["price_per_day"]
        self.distance = rental_data["distance"]
        self.numberofdays = Rental_price.count_days(rental_data["start_date"],rental_data["end_date"])
        #calculating the price
        self.price = (self.numberofdays*self.price_per_day) + (self.distance* self.price_per_km)

    def count_days(start_date,end_date):
        date = "%Y-%m-%d"
        a = datetime.strptime(start_date, date)
        b = datetime.strptime(end_date, date)
        delta = b - a
        return delta.days + 1

File: backend/level1/Rental.rb
class Rental
    attr_reader :id, :price
    def initialize(car_data,rental_data) 
        #retrieving the dictionary in cars which has the same id as the car_id from rentals
        @dictionary = car_data.select {|k| k["id"] == rental_data["car_id"]}[0]
        @id = rental_data["id"]
        #finding the next set of values needed in the price formula
        @price_per_km = @dictionary["price_per_km"]
        @price_per_day = @dictionary["price_per_day"]
        @distance = rental_data["distance"]
        @numberofdays = Rental.days(rental_data["start_date"],rental_data["end_date"])
        #calculating the price
        @price = (@numberofdays*@price_per_day) + (@distance* @price_per_km)
    end

    # creating a function to find the number of days between the start and end date
    def self.days(start_date,end_date)
        a = Date.parse(start_date)
        b = Date.parse(end_date)
        delta = b - a
        return delta.to_i + 1
    end 
end